.github/workflows/test-pipeline.yml

```diff
@@ -0,0 +1,59 @@
+name: NestJS Test Pipeline
+
+on:
+  push:
+    branches:
+      - '**'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres:15
+        ports:
+          - 5432:5432
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: task_management_test
+        options: >-
+          --health-cmd "pg_isready -U postgres"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    env:
+      STAGE: test
+      DB_HOST: localhost
+      DB_PORT: 5432
+      DB_USERNAME: postgres
+      DB_PASSWORD: postgres
+      DB_DATABASE: task_management_test
+      JWT_SECRET: KxartlkDkiOpmHwDFlPZbrtfKedeOUVmhXbnosLcJQElxiWLuY
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
```

Comment:
This is not the last stable release of the action we are using, we should always aim to use the last version of any library we decide to introduce, as long as it's compatible with the rest of the system.

Please update those!